index.jsx

```diff
@@ -1,5 +1,9 @@
 import ReactDOM from "react-dom";
 
 import App from "./App";
+import { initI18n } from "./utils/i18nUtils";
 
-ReactDOM.render(<App />, document.getElementById("app"));
+// Ensure that locales are loaded before rendering the app
+initI18n().then(() => {
```

Comment:
If I understand this correctly, this will block rendering the React tree until the locales have been fetched.   How much of a performance issue is this in development and production?  If it is a performance issue, is there any way to minimize it?